cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ConnectionHandler.java

```diff
@@ -333,11 +340,20 @@ public void spec(HttpServiceRequest request, HttpServiceResponder responder,
         ConnectorSpecRequest connectorSpecRequest = ConnectorSpecRequest.builder().setPath(specRequest.getPath())
                                                       .setConnection(connection)
                                                       .setProperties(specRequest.getProperties()).build();
-        responder.sendJson(connector.generateSpec(connectorSpecRequest));
+        ConnectorSpec spec = connector.generateSpec(connectorSpecRequest);
+        responder.sendString(GSON.toJson(getConnectorDetail(conn.getPlugin(), spec)));
       }
     });
   }
 
+  private ConnectorDetail getConnectorDetail(PluginInfo plugin, ConnectorSpec spec) {
+    Set<PluginDetail> relatedPlugins = new HashSet<>();
+    spec.getRelatedPlugins().forEach(pluginSpec -> relatedPlugins.add(
+      new PluginDetail(pluginSpec.getName(), pluginSpec.getType(), spec.getProperties(), plugin.getArtifact(),
```

Comment:
`plugin.getArtifact()` is a little bit confusing

since each PluginDetail is an entry that is related to the connector. but here the artifact is the connector's artifact instead of related plugin's artifact ?

then this information will be duplicated in the response ?

Reply:
Yea it will be duplicated, check https://github.com/cdapio/cdap/pull/13430/files#diff-1c5d74960881216e53289409ce8b2340efa1dd171f630425e28ae9941b6a9773R28-R29, this is our standard way to represent a plugin, doing it like this so UI can easily pick up these information.
The artifact for related plugins is always assumed to be in same jar with the connector